radicle/src/cob/patch.rs

```diff
@@ -849,6 +849,20 @@ impl<'a> Patches<'a> {
         Ok(state_groups)
     }
 
+    /// Find the patch containing the given `Revision`.
+    pub fn find_with_revision(
+        &self,
+        id: &RevisionId,
+    ) -> Result<Option<(PatchId, Patch)>, store::Error> {
+        let all = self.all()?;
+        let result = all
+            .into_iter()
+            .filter_map(|result| result.ok())
+            .find(|(_, p, _)| p.revisions().any(|(i, _)| i == id))
```

Comment:
I would first filter for open patches, no need to search archived patches.
Also, for patches with a single revision, we can just do `self.get(id)`; so I'd check that first, before iterating over all patches.

Reply:
> Would this not result in a 'revision not found' when someone tries merging a recently closed patch?

Yes, but you could have an error like "No open patches with revision <id> found".

> Is there overhead for 'revisions().any(..)' when there's only one revision? We'd still need to check against the revision ID.

Yes, because you might end up searching all revisions (linear time) before finding the one; vs. using `get` which is constant time.

Reply:
I feel like I'm missing something.

> > Would this not result in a 'revision not found' when someone tries merging a recently closed patch?
>
> Yes, but you could have an error like "No open patches with revision {id} found".
>
A more specific message would be helpful to the user.  How costly could the search be?  Would marking the method as 'slow' help?

> > Is there overhead for 'revisions().any(..)' when there's only one revision? We'd still need to check against the revision ID.
>
> Yes, because you might end up searching all revisions (linear time) before finding the one; vs. using `get` which is constant time.
Are other patch's revisions being included in the iteration?  I'm confused about where the overhead is coming from, because the assumption is a patch having one revision would mean one iteration.

Reply:
I would add ` Patch::revision(id: &RevisionId) -> Option<Revision>` function so we don't have to iterate over all patch revisions.

Reply:
Still hoping for this :point_up:

Reply:
[done](https://github.com/radicle-dev/heartwood/pull/461/commits/4bf75d58c0e2e7cd36f66bf9f0553a1fbf405d38) and resolved merge conflicts

Reply:
You're still iterating over all patch revisions..

Reply:
fixed